astacus/coordinator/plugins/clickhouse/plugin.py

```diff
@@ -0,0 +1,90 @@
+"""
+Copyright (c) 2021 Aiven Ltd
+See LICENSE for details
+"""
+from .config import ClickHouseConfiguration, get_clickhouse_clients, get_zookeeper_client, ZooKeeperConfiguration
+from .parts import get_frozen_parts_pattern
+from .steps import (
+    AttachMergeTreePartsStep, ClickHouseManifestStep, CreateClickHouseManifestStep, DistributeReplicatedPartsStep,
+    FreezeTablesStep, MoveFrozenPartsStep, RemoveFrozenTablesStep, RestoreAccessEntitiesStep, RestoreReplicatedDatabasesStep,
+    RetrieveAccessEntitiesStep, RetrieveReplicatedDatabasesStep, RetrieveTablesStep, SyncReplicasStep, UnfreezeTablesStep,
+    ValidateConfigStep
+)
+from astacus.common.ipc import Plugin, RestoreRequest
+from astacus.coordinator.plugins.base import (
+    BackupManifestStep, BackupNameStep, CoordinatorPlugin, ListHexdigestsStep, OperationContext, RestoreStep, SnapshotStep,
+    Step, UploadBlocksStep, UploadManifestStep
+)
+from typing import List
+
+
+class ClickHousePlugin(CoordinatorPlugin):
+    zookeeper: ZooKeeperConfiguration = ZooKeeperConfiguration()
+    clickhouse: ClickHouseConfiguration = ClickHouseConfiguration()
```

Comment:
A bit confusing yes, but a `CoordinatorPlugin` is a pydantic model and directly created from the config file, it would be strange to have "zookeeper_config" in the configuration file itself.

It can seem strange too that, in a sense, the plugin is its own config. This removes the need to write create functions to create a plugin after we have created the typed config, and it forces a pattern where the plugin `__init__` do not have side effect and can always be created without worry during tests.